### PR TITLE
Local test class names, fixed #146

### LINKED ISSUE #146: Call local test classes by their purpose - change existing section
https://github.com/SAP/styleguides/blob/master/clean-abap/CleanABAP.md#call-local-test-classes-by-their-purpose

I would like to challenge the section:

> **Call local test classes by their purpose**
>```
>CLASS ltc_unit_tests DEFINITION FOR TESTING ... .
>CLASS ltc_integration_tests DEFINITION FOR TESTING ... .
>CLASS ltc_unit_tests_with_mocks DEFINITION FOR TESTING ... .
>```
>Good names reveal the level of the tests and what's common to their setup.

👍 I agree with "Good names reveal what's common to their setup.
👎 I do not agree with "Good names reveal the level of the tests"
👎 I do not agree with the example names.

Integration vs. unit test should be 
- is it in a global test class -> integration test
- is it in a productive class -> unit test

How would you call the second and third test class? ltc_unit_tests_2?

I would rephrase it like this:

> **Call local test classes by their purpose**
> Name local test classes either by the "when" part of the story
>`CLASS ltc_<public method name> DEFINITION FOR TESTING ... ."`
> or the "given" part of the story
>`CLASS ltc_<common setup description>  DEFINITION FOR TESTING ... .`

### PR DESCRIPTION
**Pull request by thomham**

### CODE REVIEW

**Comment by HrFlorianHoffmann** [comment]
Thanks for this clarification.

**Comment by ERPvisors-SEBA** [comment]
@HrFlorianHoffmann @thomham 

The naming conventions for local test classes conflict with [Avoid Encodings](https://github.com/SAP/styleguides/blob/main/clean-abap/sub-sections/AvoidEncodings.md#reasoning) recommendations. 


```ABAP
CLASS ltc_<public method name> 
```

> With ABAP there is a legacy practice of adding prefixes to each and everything, to encode things like
>
> + cl_ for classes
> + if_ for interfaces
> + is_ for an importing parameter
> + mo_ for a class member attribute
> + lt_ for a table-like variable
> + sc_ for a constant
> + ...
> This kind of prefixing is a relic from the early days of programming, when code was printed out and read on paper, and you didn't want to flip around just to find some 
> variable's type. Modern development environments give easy access to data types, signatures, and object navigation, such that it is no longer needed to get readable code.

Or did I miss a part?
